### PR TITLE
Lyrics API

### PR DESCRIPTION
**Pull request by carolinar7**
Ended up not using Genius API, and used another API to accommodate for our needs.

### CODE REVIEW

File: src/main/webapp/script.js [hunk 1]
```diff
@@ -103,3 +103,33 @@ function getTimeAgo(timeInMilli) {
...
+  // Calls lyrics.ovh API
+  request.open('GET', 'https://api.lyrics.ovh/v1/' + artist + '/' + song);
+  // Returns lyrics in JSON format
+  request.onreadystatechange = function () {
+    if (this.readyState === 4) {
+      var lyricsJSON = this.responseText;
+      // API will return two key values, error (if not found), 
+      // or lyrics (if found, but could possibly return empty string)
+      obj = JSON.parse(lyricsJSON);
+      if (obj.lyrics && obj.lyrics != '' && obj.lyrics != null) {
```
Comment (mariah-crawford) [inline src/main/webapp/script.js]: I suggest adding parentheses to separate the conditions, such as around the conditions that come before and after the second "&&", in order to add clarity on the use of the logical operators.

File: src/main/webapp/script.js [hunk 2]
```diff
@@ -103,3 +103,33 @@ function getTimeAgo(timeInMilli) {
...
+  // Gets value from inputs and sets them
+  var artist = document.getElementById('artist').value;
+  var song = document.getElementById('song').value;
+  var request = new XMLHttpRequest();
+  var obj;
+  // Calls lyrics.ovh API
+  request.open('GET', 'https://api.lyrics.ovh/v1/' + artist + '/' + song);
+  // Returns lyrics in JSON format
+  request.onreadystatechange = function () {
+    if (this.readyState === 4) {
```
Comment (mariah-crawford) [inline src/main/webapp/script.js:118]: I am curious about the use of an explicit number, could you explain what this line checks/verifies, maybe via a comment?

File: src/main/webapp/script.js [hunk 3]
```diff
@@ -103,3 +103,33 @@ function getTimeAgo(timeInMilli) {
...
+      var lyricsJSON = this.responseText;
+      // API will return two key values, error (if not found), 
+      // or lyrics (if found, but could possibly return empty string)
```
Comment (mariah-crawford) [inline src/main/webapp/script.js:121]: Nice use of descriptive comments here :)

File: src/main/webapp/script.js [hunk 4]
```diff
@@ -103,3 +103,33 @@ function getTimeAgo(timeInMilli) {
...
+      obj = JSON.parse(lyricsJSON);
+      if (obj.lyrics && obj.lyrics != '' && obj.lyrics != null) {
+        // Call function that does sentiment analysis on obj.lyrics string
+        document.getElementById("form").submit();
+      } else {
+        document.getElementById('error-lyric').innerHTML = 'Could not find song.' + 
+          ' Check spelling, try to submit again, or choose another song.';
```
Comment (mariah-crawford) [inline src/main/webapp/script.js:128]: Quick fix: could you delete the space before "Check"?

File: src/main/webapp/script.js [hunk 5]
```diff
@@ -103,3 +103,33 @@ function getTimeAgo(timeInMilli) {
...
+  var artist = document.getElementById('artist').value;
+  var song = document.getElementById('song').value;
+  var request = new XMLHttpRequest();
+  var obj;
+  // Calls lyrics.ovh API
+  request.open('GET', 'https://api.lyrics.ovh/v1/' + artist + '/' + song);
+  // Returns lyrics in JSON format
+  request.onreadystatechange = function () {
+    if (this.readyState === 4) {
+      var lyricsJSON = this.responseText;
```
Comment (mariah-crawford) [inline src/main/webapp/script.js]: Could you potentially rename this variable to lyricsJson using the similar variable format seen above?

File: src/main/webapp/index.html [hunk 6]
```diff
@@ -21,15 +21,18 @@ <h3>Quote of the Day</h3>
...
         <p>Enter your mood on a scale from 1-10:</p>
         <input type="number" name="mood" min="1" max="10" value="1">
         <p>Enter a song name</p>
-        <textarea name="song" maxlength="50" rows="1" cols="50" placeholder="Song name" required></textarea>
+        <textarea id="song" name="song" maxlength="50" rows="1" cols="50" placeholder="Song name" required></textarea>
         <p>Enter the artist name</p>
-        <textarea name="artist" maxlength="100" rows="1" cols="50" placeholder="Artist name" required></textarea>
+        <textarea id="artist" name="artist" maxlength="100" rows="1" cols="50" placeholder="Artist name" required></textarea>
+        <p>Enter your mood on a scale from 1-10:</p>
+        <input type="number" name="mood" min="1" max="10" value="1">
```
Comment (mariah-crawford) [inline src/main/webapp/index.html:32]: There is code duplication here for the p and input elements that correspond to the mood as seen above, could you delete this please?

File: src/main/java/com/google/sps/servlets/DataServlet.java [hunk 7]
```diff
@@ -69,7 +69,6 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     // Get the input and time from the form.
     String textEntryString = request.getParameter("text");
     String moodScaleString = request.getParameter("mood");
-    String songEntryString = request.getParameter("song");
```
Comment (mariah-crawford) [inline src/main/java/com/google/sps/servlets/DataServlet.java:72]: Could you add this back to the code, I see it is later referenced when storing the song to Datastore via the journalEntity?

**Review by mariah-crawford — CHANGES_REQUESTED** [review]
Overall nice work, especially with discovering a different API that accommodates our needs! I like the use of comments, which made the code more understandable. :) There are some quick fixes that I describe more in depth in my review comments.

File: src/main/webapp/script.js (hunk 2 above)
Comment (carolinar7) [reply src/main/webapp/script.js:118]: It's from the API documentation. I'm not a hundred percent sure, but as the name kind of suggests and with some guessing, it verifies that the lyrics are ready to be processed/used.

File: src/main/webapp/script.js (hunk 1 above)
Comment (carolinar7) [reply src/main/webapp/script.js]: You're right, can get a little confusing, I fixed it.

File: src/main/webapp/script.js (hunk 3 above)
Comment (carolinar7) [reply src/main/webapp/script.js:121]: Thanks!

File: src/main/webapp/script.js (hunk 4 above)
Comment (carolinar7) [reply src/main/webapp/script.js:128]: The space before "Check", is used to properly space the sentence.
With space: Could not find song. Check spelling, try to submit again, or choose another song.
Without space: Could not find song.Check spelling, try to submit again, or choose another song.

File: src/main/webapp/script.js (hunk 5 above)
Comment (carolinar7) [reply src/main/webapp/script.js]: fixed!

File: src/main/webapp/index.html (hunk 6 above)
Comment (carolinar7) [reply src/main/webapp/index.html:32]: I think I messed up when I merged in master

File: src/main/java/com/google/sps/servlets/DataServlet.java (hunk 7 above)
Comment (carolinar7) [reply src/main/java/com/google/sps/servlets/DataServlet.java:72]: I think I messed up when I merged in master

**Comment by carolinar7** [comment]
@mariah-crawford take a look at the changes!

**Review by mariah-crawford — APPROVED** [review]
The changes look good!

**Pull request merged**